Mykhailenko_Homework_9_date_20-03-20/GroupsOrganization/UnivercityLists.cs

```diff
@@ -0,0 +1,190 @@
+        public static SciStaff GetSciWorker(int index)
+        {
+            return SciStaffList[index];
+        }
+        public static Student GetStudent(int index)
+        {
+            return StudentsList[index];
+        }
+
+        public static string GetRandomName()
+        {
+            return Enum.GetName(typeof(Names), Random.Next(Enum.GetValues(typeof(Names)).Length));
+        }
+        public static string GetRandomSurName()
+        {
+            return Enum.GetName(typeof(Surnames), Random.Next(Enum.GetValues(typeof(Surnames)).Length));
+        }
+
+        public static string DefaultStudyGroupsFilling()
+        {
+            int currentGroup = 0;
+            int currentStudent = 0;
+            string result = string.Empty;
+
+            while ((currentStudent < StudentsList.Count) && (currentGroup < SciStaffList.Count))
+            {
+                SciStaffList[currentGroup].AddStudentToGroup(StudentsList[currentStudent++]);
+
+                if (SciStaffList[currentGroup].SupervizedGroup.Count >= SciStaffList[currentGroup].GetMaxGroupCapacity())
+                {
+                    result += $"{SciStaffList[currentGroup].GetType().Name} {SciStaffList[currentGroup].Name} {SciStaffList[currentGroup].SurName}'s group was filled.\n";
+                    currentGroup++;
+                }
+            }
+
+            return result;
+        }
+
+        public static string SciStaffGroupsInfoPrint()
+        {
+            string result = string.Empty;
+
+            int groupCounter = 1;
+
+            foreach (SciStaff item in SciStaffList)
+            {
+                result +=
+                    $"Group #{groupCounter++}\n" +
+                    item.ToString() + "\n";
+            }
+
+            return result;
+        }
+        public static string StudentsBindingInfoPrint()
+        {
+            string result = string.Empty;
+
+            foreach (Student item in StudentsList)
+            {
+                result += item.ToString() + "\n";
+            }
+
+            return result;
+        }
+
+        public static string GetNonEmptyGroupsInfo()
+        {
+            string result = string.Empty;
+            int groupCounter = 1;
+
+            foreach (SciStaff item in SciStaffList.Where(teacher => teacher.SupervizedGroup.Count != 0).ToList())
+            {
+                result +=
+                    $"Group #{groupCounter++}\n" +
+                    $"Group lead: {item.GetType().Name.ToLower()} {item.Name} {item.SurName} " +
+                    $"Group fullness: {item.SupervizedGroup.Count}/{item.GetMaxGroupCapacity()}\n";
+            }
+
+            return result;
+        }
+
+        public static int GetNonEmptyGroupsCount()
+        {
+            return SciStaffList.Where(teacher => teacher.SupervizedGroup.Count != 0).ToList().Count;
+        }
+        public static string GetDetailedGroupInfo(int index)
+        {
+
+            string result = string.Empty;

```

Comment:
try to substitute it with StringBuilder class. Just for practice)
https://www.educba.com/string-vs-stringbuilder/